packages/spy/src/index.ts

```diff
@@ -104,65 +105,72 @@ type Classes<T> = {
   [K in keyof T]: T[K] extends new (...args: any[]) => any ? K : never
 }[keyof T] & (string | symbol)
 
-/**
- * @deprecated Use MockInstance<A, R> instead
- */
-export interface SpyInstance<TArgs extends any[] = any[], TReturns = any> extends MockInstance<TArgs, TReturns> {}
+/*
+cf. https://typescript-eslint.io/rules/method-signature-style/
 
-export interface MockInstance<TArgs extends any[] = any[], TReturns = any> {
+Typescript assignability is different between
+  { foo: (f: T) => U } (this is "method-signature-style")
+and
+  { foo(f: T): U }
+
+Jest uses the latter for `MockInstance.mockImplementation` etc... and it allows assignment such as:
+  const boolFn: Jest.Mock<() => boolean> = jest.fn<() => true>(() => true)
+*/
```

Comment:
I took the latter one to follow Jest for now, but I think going with the former should be totally fine too.
Jest one is technically type unsafe and the assignment example I wrote here can be easily rewritten by:

```ts
jest.fn<() => boolean>(() => true)
jest.fn(() => true as boolean)
```

Or in a more realistic scenario, it would probably look like:

```ts
// setup
type MyFn = () => boolean;
let myMockFn: Mock<MyFn>

// test
myMockFn = vi.fn<MyFn>(() => true)
```